src/mdformat/renderer/__init__.py

```diff
@@ -100,8 +101,16 @@ def render_tree(
 
     @staticmethod
     def _write_references(env: MutableMapping) -> str:
+        def label_sort_key(label: str) -> str:
+            if not label:
+                raise AssertionError("link label cannot be empty string")
```

Comment:
```suggestion
            assert label, "link label cannot be empty string"
```
This is better because it get optimized away with `python -O`

Reply:
Yep! I try to avoid using `assert` for runtime validation for that reason.

Reply:
IMO it's desirable here because the `if not label` condition is a something that should never happen unless there's a programming error. A label cannot be an empty string in CommonMark. The check is mostly there for peace of mind and to prove the programmer that the `if not label` condition is never true (if it were, it'd have implications for the `all()` calll that follows).